tests/regression_tests/regression/test_springs_force.cpp

```diff
@@ -0,0 +1,97 @@
+#include <gtest/gtest.h>
+
+#include "test_utilities.hpp"
+
+#include "src/elements/springs/create_springs.hpp"
+#include "src/elements/springs/springs.hpp"
+#include "src/model/model.hpp"
+#include "src/step/update_system_variables_springs.hpp"
+
+namespace openturbine::tests {
+
+inline auto SetUpSpringsForceTest() {
+    auto model = Model();
+
+    // Add two nodes for the spring element
+    model.AddNode(
+        {0, 0, 0, 1, 0, 0, 0},  // First node at origin
+        {0, 0, 0, 1, 0, 0, 0}, {0, 0, 0, 0, 0, 0}, {0, 0, 0, 0, 0, 0}
+    );
+    model.AddNode(
+        {1, 0, 0, 1, 0, 0, 0},  // Second node at (1,0,0)
+        {0, 0, 0, 1, 0, 0, 0}, {0, 0, 0, 0, 0, 0}, {0, 0, 0, 0, 0, 0}
+    );
+
+    const auto springs_input = SpringsInput({SpringElement(
+        std::array{model.GetNode(0), model.GetNode(1)},
+        10.,  // Spring stiffness coeff.
+        1.    // Undeformed length
+    )});
+
+    auto springs = CreateSprings(springs_input);
+    auto state = model.CreateState();
+
+    return std::make_tuple(springs, state);
+}
+
+TEST(SpringsForceTest, ZeroDisplacement) {
+    auto [springs, state] = SetUpSpringsForceTest();
+
+    UpdateSystemVariablesSprings(springs, state);
+
+    auto l_ref_host = Kokkos::create_mirror(springs.l_ref);
+    auto l_host = Kokkos::create_mirror(springs.l);
+    auto c1_host = Kokkos::create_mirror(springs.c1);
+    auto f_host = Kokkos::create_mirror(springs.f);
+    auto a_host = Kokkos::create_mirror(springs.a);
+
+    Kokkos::deep_copy(l_ref_host, springs.l_ref);
+    Kokkos::deep_copy(l_host, springs.l);
+    Kokkos::deep_copy(c1_host, springs.c1);
+    Kokkos::deep_copy(f_host, springs.f);
+    Kokkos::deep_copy(a_host, springs.a);
+
+    EXPECT_DOUBLE_EQ(l_ref_host(0), 1.);  // Undeformed length = 1.
+    EXPECT_DOUBLE_EQ(l_host(0), 1.);      // Current length = 1.
+    EXPECT_DOUBLE_EQ(c1_host(0), 0.);     // c1 = 0.
+    expect_kokkos_view_1D_equal(Kokkos::subview(f_host, 0, Kokkos::ALL), {0., 0., 0.});  // No force
```

Comment:
this doesn't compile on GPU.  all 'expect_kokkos_view_*D_equal` functions require a view in the default memory space.  For this case, pass 'springs.f` instead of `f_host`.  The same will need to be done on line 59 and 89 in this file.